firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -243,7 +251,7 @@ private final void doRegistration() {
 
       // FID needs to be registered
       if (persistedInstallationEntry.isUnregistered()) {
-        registerAndSaveFid(persistedInstallationEntry);
+        registerAndSaveFid(persistedInstallationEntry, iidToken);
```

Comment:
does this mean we are always sending the IID-Token even if its not a migration of an Instance-ID?

Reply:
No, just for simplifying the code change I am passing the IID-Token (null or  the token read from iid shared pref) .

When calling FIS createInstallation I check if IID-Token is null, I skip setting the header.
Refer to:
https://github.com/firebase/firebase-android-sdk/pull/980/files#r347046883